Qixiang feat on netanel 0610

qixiang-99 opened the pull request:
# PR title

Please write the PR title by following template:

[JIRA ticket link/nvbug link/github issue link][fix/feat/doc/infra/...] \<summary of this PR\>

For example, assume I have a PR hope to support a new feature about cache manager of Jira TRTLLM-1000 ticket, it would be like

[TRTLLM-1000][feat] Support a new feature about cache manager

## Description

Please explain the issue and the solution in short.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

netanel-haber committed:
allocate blocks per window size correctly

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

netanel-haber committed:
simpler code path for common homogeneous models

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
shorten: (b|B)locksPerWindowSize -> blocksPerWindow

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
fix trivial test compile errors

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
fix non-trivial compile errors

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
fix resource manager

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
fix extracostmemory

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

qixiang-99 committed:
Refactor and pipeclean, can pass gemma3 test. Need to run CI to find out whether there are more issues.

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>

netanel-haber committed:
minimize diff

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
small fix

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
dynamic batch tuning

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix tests

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix blocks_per_window

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
use windowSizeToLayers for improved clarity instead of managedLayers for clarity

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
docs and naming

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
provide free memory to calculateMaxNumBlocks as an argument, so cross kvcache fractioning can be done correctly

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
remove unused imports

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
hopefully implement mpi sync

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
only warn when VSWA + config.maxTokens is set

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
clamp maxAttentionWindowVec

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix test

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
better logs

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix windowSizeToBlocks indexing

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix KVCacheManagerLeafBlockWithDependentTest

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix WindowSizeMetadata fields ordering

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix KVCacheManagerVariableWindowAttentionWithReuseTest

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
Changed type for maxTokens to uint64_t to avoid overflow

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
*multiply* by crossKvCacheFraction, not divide

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix cross manager window size

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix test_KvCache_events_binding kvcachemanager init

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
fix calculate_max_num_blocks binding

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
assert freeMemory smaller than totalMemory

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
assert freeMemory smaller than totalMemory - after printing them

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
metadata.allottedPrimaryBlocks / blockRequirementsPerSequence instead of blockRequirementsPerSequence / metadata.allottedPrimaryBlocks; The window with the *smallest* max batch size is the limiting factor, hence, the std::*min* of all the max batch sizes is chosen: minMaxBatchSizeAllWindows = std::min(minMaxBatchSizeAllWindows, maxBatchSizeWindow)

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

qixiang-99 committed:
fix:
- Many tests directly pass executor.KvCacheConfig to KvCacheManager, apply a conversion if this happen
- add `self.blocks_in_primary_pool` and `self.blocks_in_secondary_pool` as these are used for standard case.

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>

chzblych committed:
[Infra] - Update JNLP container config (#5008)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

netanel-haber committed:
fix minor bug

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
actually use reduced value [blocksWorld] and assign it to blocksPrimary and blocksSecondary

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

qixiang-99 committed:
fix: correct the test that passes the kvcache config with the wrong type (the mirror class of executor.kvcache_config)

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

netanel-haber committed:
pr comments

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
make logging quieter

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
add ceremony

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #8352](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8352/) [ run ] triggered by Bot

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

tensorrt-cicd commented:
[PR_Github #8352](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8352/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #6047](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/6047/) completed with status: 'FAILURE'

qixiang-99 committed:
fix:
- relax the type restirction for kv_cache_config and only require as needed.
- update get_cache_indices and get_batch_cache_indices method accordingly
- revert the change of test

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>
